src/pricing-helper.ts

```diff
@@ -0,0 +1,157 @@
+import {
+  Address,
+  LoggerConstructor,
+  Logger,
+  Token,
+  PoolPrices,
+  ExchangePrices,
+} from './types';
+import {
+  SwapSide,
+  SETUP_RETRY_TIMEOUT,
+  FETCH_POOL_INDENTIFIER_TIMEOUT,
+  FETCH_POOL_PRICES_TIMEOUT,
+} from './constants';
+import { DexAdapterService } from './dex';
+
+export class PricingHelper {
+  logger: Logger;
+
+  constructor(
+    protected dexAdapterService: DexAdapterService,
+    loggerConstructor: LoggerConstructor,
+  ) {
+    this.logger = loggerConstructor(
+      `PricingHelper_${dexAdapterService.network}`,
+    );
+  }
+
+  private async initializeDex(dexKey: string, blockNumber: number) {
+    try {
+      const dexInstace = this.dexAdapterService.getDexByKey(dexKey);
```

Comment:
`dexInstace` -> `dexInstance` (also in `getPoolIdentifiers`)

Reply:
Fixed